src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,73 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        List<CustomerSalary> customerSalaryList = new ArrayList<>();
+
+        for (String record : data) {
+            String[] recordArr = record.split(" ");
+            String recordDate = recordArr[0];
+            String name = recordArr[1];
+            String hours = recordArr[2];
+            String rate = recordArr[3];
```

Comment:
```suggestion
            String recordDate = recordArr[DATE_INDEX];
            String name = recordArr[NAME_INDEX];
            ....
```

numbers 0, 1, 2, ... should be constants with informative names